cmd/adsysd/client/policy.go

```diff
@@ -186,6 +187,15 @@ func (a *App) dumpPolicies(target string, showDetails, showOverridden, nocolor b
 			return fmt.Errorf("failed to retrieve current user: %w", err)
 		}
 		target = u.Username
+
+		if isMachine {
+			hostname, err := os.Hostname()
+			if err != nil {
+				return fmt.Errorf("failed to retrieve client hostname: %w", err)
+			}
+			// For bad instances where /etc/hostname returns FQDN.
+			target, _, _ = strings.Cut(hostname, ".")
```

Comment:
We fixed it in the ad instances... The problem is that we don't have access to it in this context IIRC, so there was no other choice.

Reply:
Did you find you needed it when testing manually?

Now that we added normalizing the target name in `func (ad *AD) NormalizeTargetName(ctx context.Context, target string, objectClass ObjectClass) (string, error) {` for any target we get into the server, I don’t think this is needed. If it’s still needed because you saw this in your tests, I think the bug is elsewhere and we should fix it.

If you only added it because it was already in `func (a *App) update(isComputer, updateAll bool, target, krb5cc string) error {`, I think stripping it over there too should be safe, but needs testing ofc first! (If it’s copied though, I wonder why the comment iis different from the other one)

Reply:
I tested it manually now and it it's not needed, indeed. Since I was using `os.Hostname()`, I expected it to bypass our treatment of FQDNs, but before dumping the policies there is a call to NormalizeTargetName, so it's fine. Since I wasn't 100% sure about the flow of gRPC DumpPolicies service, I pre added this there, but it's removed now.